tcms/rpc/tests/test_testplan.py

```diff
@@ -226,3 +222,114 @@ def test_ignores_existing_mappings(self):
                         case=case_id
                     ).count()
                 )
+
+
+@override_settings(LANGUAGE_CODE='en')
+class TestCreate(APITestCase):
+    def test_create_plan_with_empty_required_field(self):
+        product = ProductFactory()
+        version = VersionFactory(product=product)
+        plan_type = PlanTypeFactory()
+        with self.assertRaisesRegex(XmlRPCFault, 'This field is required.'):
+            self.rpc_client.TestPlan.create({
+                'product': product.pk,
+                'product_version': version.pk,
+                'name': '',
+                'type': plan_type.pk,
+                'text': 'Testing TCMS',
+                'parent': None,
+            })
+
+
+class TestCreatePermission(APIPermissionsTestCase):
+    permission_label = 'testplans.add_testplan'
+
+    def _fixture_setup(self):
+        super()._fixture_setup()
+
+        self.product = ProductFactory()
+        self.version = VersionFactory(product=self.product)
+        self.plan_type = PlanTypeFactory()
+        self.params = {
+            'product': self.product.pk,
+            'product_version': self.version.pk,
+            'name': 'Testplan foobar',
+            'type': self.plan_type.pk,
+            'text': 'Testing TCMS',
+            'parent': None,
+        }
+
+    def verify_api_with_permission(self):
+        result = self.rpc_client.TestPlan.create(self.params)
+        self.assertEqual(self.params['product'], result['product_id'])
+        self.assertEqual(self.params['product_version'], result['product_version_id'])
+        self.assertEqual(self.params['name'], result['name'])
+        self.assertEqual(self.params['type'], result['type_id'])
+        self.assertEqual(self.params['text'], result['text'])
+        self.assertEqual(self.params['parent'], result['parent'])
```

Comment:
Also assert that author is the current user sending the request. See lines 60-61 in the method under test.